src/tokenize_cn.c

```diff
@@ -62,31 +91,70 @@ static uint32_t cnTokenizer_Next(RSTokenizer *base, Token *t) {
     }
 
     switch (tok->type) {
-      // Skip words we know we don't care about.
+        // Skip words we know we don't care about.
       case __LEX_STOPWORDS__:
-      case __LEX_PUNC_WORDS__:
       case __LEX_ENPUN_WORDS__:
       case __LEX_CJK_UNITS__:
       case __LEX_NCSYN_WORDS__:
+        inEscape = 0;
         continue;
+
+      case __LEX_PUNC_WORDS__:
+        if (inEscape && istoksep(tok->word[0])) {
+          appendToEscbuf(self, tok->word, tok->length);
```

Comment:
You are not checking the return value, it might be that the value was not totally added cause buff is full. If you do not think it will ever happened then put assert. Otherwise handle the error.